massgov/openmass
# DP-22196: SPIKE Assess language negotiation capability in Drupal with language module

**Description:**
Added browser language negotiation and redirection for nodes with translations.


**Jira:** (Skip unless you are MA staff)
https://jira.mass.gov/browse/DP-22196


**To Test:**
You will need to use a browser extension to modify your browser language for testing. Here is a chrome extension that works well: https://chrome.google.com/webstore/detail/locale-switcher/kngfjpghaokedippaapkfihdlmmlafcc

- With your browser set to English (default), go to /forms/formulario-de-recertificacao-de-erap.
- [ ] Verify you are redirected to the English version: /forms/emergency-rental-assistance-recertification-application.
- Change your browser language to "Portuguese (Brazil)" or "pt-br" and refresh the English page: /forms/emergency-rental-assistance-recertification-application.
- [ ] Verify you are redirected to the Portuguese (Brazil) version: /forms/emergency-rental-assistance-recertification-application.
- Go to /how-to/file-an-appeal-with-dta (or a page without translations).
- [ ] Verify there is no redirection.
- Reset your browser language to English (default).
- Login and go to /forms/emergency-rental-assistance-recertification-application/translations.
- [ ] Verify the page loads as expected in comparison with Prod.
- Go to /doc/massdot-hearing-handout-boston-neponset-river-greenway-052621-portuguese/translations.
- [ ] Verify the page loads as expected in comparison with Prod.

**Screenshots/GIFs:**

---

[Peer Review Checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md)
